### PR TITLE
refactor(markdown): removed `rehype-slug` in favor of our own implementation

### PR DESCRIPTION
**Pull request by JuanM04**
## Changes

While reviewing #3044, I've noticed that there is a function (`rehypeCollectHeaders`) which is being used to generate slugs no matter what plugins the user specified. So, I've removed `rehype-slug` because it was doing the exact same thing (it even uses the same slugger!).

Sidenote: I've moved some types around, so `headers` are typed instead of `any[]`

## Testing

It's already being tested

## Docs

TDB

### CODE REVIEW

File: packages/markdown/remark/src/rehype-collect-headers.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
-/**  */
 export default function createCollectHeaders() {
-	const headers: any[] = [];
+	const headers: MarkdownHeader[] = [];
+	const slugger = new Slugger();
 
-	function rehypeCollectHeaders() {
-		return function (tree: any) {
+	const rehypeCollectHeaders: RehypePlugin<[]> = () => {
```
Comment (FredKSchott) [inline packages/markdown/remark/src/rehype-collect-headers.ts:10]: style nit: we usually avoid `const X = () => {}` in this code base, in favor of `function X() {}`. 

Unless I'm missing something, you should be able to get the same type info like this:

```
function rehypeCollectHeaders(): RehypePlugin<[]> {
```

**Review by FredKSchott — APPROVED** [review]
LGTM with 1 small style comment! Merge when ready

File: packages/markdown/remark/src/rehype-collect-headers.ts (hunk 1 above)
Comment (JuanM04) [reply packages/markdown/remark/src/rehype-collect-headers.ts]: Updated that!

**Comment by FredKSchott** [comment]
Feel free to merge if you can confirm that this PR didn't introduce this issue: https://github.com/withastro/docs/pull/404#issuecomment-1113145885

**Comment by JuanM04** [comment]
@FredKSchott ok, so I've added some notes about how it currently works (with or without this PR) in the documentation:

> Astro will add autogenerated ids to all headings in Markdown files automatically using [github-slugger](https://github.com/Flet/github-slugger) — if a custom id is specified, it won't be overriden. These ids will be added _after_ all the other plugins are executed, so if you have a plugin like `rehype-toc` that needs ids, you should add your own slugging plugin (like `rehype-slug`).

**Review by FredKSchott — APPROVED** [review]
LGTM

**Pull request merged**